File: 0069 sqrtx/solution.go
func mySqrt(x int) int {
    low,high:=0,x
    for low<=high {
        mid:=(low+high)/2
        if mid*mid==x {
            return mid
        }   else if mid*mid<x {
            low=mid+1
        } else {
            high = mid-1
        }
    }
    return high
}

File: 0069 sqrtx/solution.ts
function mySqrt(x: number): number {
    let l = 0,r = x;
    while(l<=r) {
        let mid = Math.floor(l+(r-l)/2);
        let sq = mid*mid;
        if(sq==x) return mid;
        else if(sq<x) l=mid+1;
        else r = mid-1;
    }
    return r;
};
